util/common-config/jest-common.js

```diff
@@ -55,6 +59,14 @@ module.exports = {
                 ],
             ],
             globals: {
+                "ts-jest": {
+                    // js-with-ts preset require "allowJs": true in tsconfig.json
+                    // so making a "tsconfig.test.json" to override it
+                    tsconfig: "config/tsconfig.test.json",
```

Comment:
Can we either change the `createConfig()` function to not take in a tsconfig, or pass `config/tsconfig.test.json` in for each package?

Reply:
We are passing a `tsconfig` object read from `tsconfig.json` for each package, and the only usage of this object is to pass `compilerOptions.paths` into jest's `moduleMapper`. But I didn't find any `tsconfig.json` we read from has the `compilerOptions.paths` field, do you think it's OK to remove the `tsconfig` object argument for `createConfig()`?

Reply:
It looks like `ts-jest`'s tsconfig property supports an inline object, so I guess I'd lean toward using the passed-in object. Then we'd just have to update each jest.config.js from `require("./tsconfig.json")` to `require("./config/tsconfig.test.json")`. That also keeps the `config/ts*.json` directory structure a convention rather than something jest-common.js needs to work.

Reply:
Removed `tsconfig` parameter for `createConfig()`